string type
---
bool is_interaction_successful
---